### PR TITLE
ci:  fix dependabot can not access the secrets

### LINKED ISSUE #4761: Dependabot can not access the secrets
**What would you like to be added**:

/assign

I really tested the dependabot on my fork, and skiped some CI due to `if: ${{ github.repository == 'karmada-io/karmada' }}`.

Now, have two case need to resolve,

##  FOSSA failed 
https://github.com/karmada-io/karmada/blob/ff7322acf252632ef2c20864b904a6cd65fa5400/.github/workflows/fossa.yml#L18

It can be resolved by add a secrets of dependabot. cc @RainbowMango 
image

## upload sarif to codeql failed

https://github.com/karmada-io/karmada/blob/ff7322acf252632ef2c20864b904a6cd65fa5400/.github/workflows/ci-image-scanning.yaml#L47-L50

> https://github.com/karmada-io/karmada/actions/runs/8445321687/job/23132353774?pr=4753#step:7:64
> Warning: Workflows triggered by Dependabot on the "push" event run with read-only access. Uploading Code Scanning results requires write access. To use Code Scanning with Dependabot, please ensure you are using the "pull_request" event for this workflow and avoid triggering on the "push" event for Dependabot branches. See https://docs.github.com/en/code-security/secure-coding/configuring-code-scanning#scanning-on-push for more information on how to configure these events.

Follow the warning, let this CI working on pull_request for dependabot to resolve it.

verifying at https://github.com/liangyuanpeng/karmada/actions/runs/8460139871/job/23177775457?pr=53

**Why is this needed**:

### PR DESCRIPTION
**Pull request by liangyuanpeng**
**What type of PR is this?**

/kind cleanup
**What this PR does / why we need it**:

PartOf https://github.com/karmada-io/karmada/issues/4761

skip the push event for dependabot

**Which issue(s) this PR fixes**:
Fixes #4761

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: .github/workflows/ci-image-scanning.yaml [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 on:
   push: 
+  pull_request:
 jobs:
   use-trivy-to-scan-image:
     name: image-scanning
-    if: ${{ github.repository == 'karmada-io/karmada' }}
+    # Trigger by pull_request to work for dependabot, for running github/codeql-action/upload-sarif
+    # Otherwise trigger by push
+    if: ( github.repository == 'karmada-io/karmada' && github.actor == 'dependabot[bot]' && github.event_name == 'pull_request') || ( github.repository == 'karmada-io/karmada' && github.actor != 'dependabot[bot]' && github.event_name == 'push' )
```
Comment (RainbowMango) [inline .github/workflows/ci-image-scanning.yaml:10]: Feels a little bit tricky, is it the recommended way?

Comment (liangyuanpeng) [reply .github/workflows/ci-image-scanning.yaml:10]: I understand that expressions is look a little scary, and we have other two option:

1. only trigger this workflow on push:

```diff
on:
--  push: 
++  pull_request:
```

2. run with two job, the first job working for push and the second job working for dependabot & pull_request

```yaml
on:
  push:
  pull_request:
  
jobs:
  use-trivy-to-scan-image:
    name: image-scanning
    if:  github.repository == 'karmada-io/karmada' && github.actor == 'dependabot[bot]' && github.event_name == 'pull_request'

  use-trivy-to-scan-image-bot:
  if: github.repository == 'karmada-io/karmada' && github.actor != 'dependabot[bot]' && github.event_name == 'push'
...
```

I would suggest option 1, which is simple enough and meets the requirements.

Comment (RainbowMango) [reply .github/workflows/ci-image-scanning.yaml:10]: > only trigger this workflow on push:

Do you mean on `pull_request`?

Now the workflow is exactly [on-push](https://github.com/karmada-io/karmada/blob/c310f89c126db93fd722447c21962201b071f1e3/.github/workflows/ci-image-scanning.yaml#L3).

**Comment by liangyuanpeng** [comment]
> https://github.com/github/codeql-action/pull/435#issuecomment-810350417
> Although perhaps you could add "Uploading code scanning results requires write access." as second sentence?

I found the PR for this warnning log and seems like it just wanted to draw people's attention to the fact that this requires write permissions.

So just add permission for this workflow to resolve it, keeping  on push event to trigger it.

PTAL,Thanks.
/cc zhzhuang-zju @RainbowMango

**Comment by liangyuanpeng** [comment]
And it's verifyed on my fork repo, check https://github.com/liangyuanpeng/karmada/actions/runs/8518835905/job/23331711824?pr=53

**Comment by zhzhuang-zju** [comment]
> I found the PR for this warnning log and seems like it just wanted to draw people's attention to the fact that this requires write permissions.
>

@liangyuanpeng Thanks for your work! This warning log is indeed very confusing. Since it's just a permissions issue, I think your change is appropriate.

**Comment by liangyuanpeng** [comment]
PTAL,Thanks.

@RainbowMango @zhzhuang-zju

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**